storage/badger/receipts.go

```diff
@@ -140,16 +140,29 @@ func (r *ExecutionReceipts) Index(blockID, receiptID flow.Identifier) error {
 		}
 
 		return nil
-	})
+	}
+	r.byBlockId = func(tx *badger.Txn, blockID flow.Identifier) ([]*flow.ExecutionReceipt, error) {
+		var receiptID flow.Identifier
+		err := operation.LookupExecutionReceipt(blockID, &receiptID)(tx)
+		if err != nil {
+			return nil, fmt.Errorf("could not lookup receipt ID: %w", err)
+		}
+		receipt, err := r.byID(receiptID)(tx)
+		if err != nil {
+			return nil, err
+		}
+		return []*flow.ExecutionReceipt{receipt}, nil
```

Comment:
From [this part](https://github.com/onflow/flow-go/pull/486/files#diff-d8547dad4f02ecbdc866e2778645382aa5cd89ba25a5f9441cc0e0c094218429R36-R43) it seems that possible to return an _empty_ slice even when our error is `nil`. If that is the case, I would do the check here and return a _not found error_, so that we do not need any extra checking externally other than checking the error itself.

Reply:
Basically did that to ensure that it's not nil. I am not sure how should we behave.
Suppose we don't have any receipts for `blockA`, should we return `([], nil)` or `(nil, err.NotFound)`.
That's an open question.

Reply:
- When returning error then you can be sure that content is nil and you can't use it, can't do smth like `receipts[0]`. But when error is nil you can be safe to get first receipt without checking for `len()`.
- When there is no error you need to check if slice is not empty before accessing it.